tensorflow/tools/ci_build/ci_parameterized_build.sh

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env bash
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+#
+# Usage:
+#   ci_parameterized_build.sh
+#
+# The script obeys the following required environment variables:
+#   TF_BUILD_CONTAINER_TYPE:   (CPU | GPU | ANDROID)
+#   TF_BUILD_PYTHON_VERSION:   (PYTHON2 | PYTHON3)
+#   TF_BUILD_IS_OPT:           (NO_OPT | OPT)
+#   TF_BUILD_IS_PIP:           (NO_PIP | PIP)
+#
+# Note: certain combinations of parameter values are regarded
+# as invalid and will cause the script to exit with code 0. For example:
+#   NO_OPT & PIP     (PIP builds should always use OPT)
+#   ANDROID & PIP    (Android and PIP builds are mutually exclusive)
+#
+# Additionally, the script follows the directions of optional environment
+# variables:
+#   TF_BUILD_DRY_RUN:  If it is set to any non-empty value that is not "0",
+#                      the script will just generate and print the final
+#                      command, but not actually run it.
+#   TF_BUILD_APPEND_CI_DOCKER_EXTRA_PARAMS:
+#                      String appended to the content of CI_DOCKER_EXTRA_PARAMS
+#   TF_BUILD_APPEND_ARGUMENTS:
+#                      Additional command line arguments for the bazel,
+#                      pip.sh or android.sh command
+#   TF_BUILD_BAZEL_TARGET:
+#                      Used to override the default bazel build target:
+#                      //tensorflow/...
+#
+# This script can be used by Jenkins parameterized / matrix builds.
+
+# Helper function: Convert to lower case
+to_lower () {
+  echo "$1" | tr '[:upper:]' '[:lower:]'
+}
+
+# Helper function: Strip leading and trailing whitespaces
+str_strip () {
+  echo -e "$1" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//'
+}
+
+
+##########################################################
+# Default configuration
+CI_BUILD_DIR="tensorflow/tools/ci_build"
+
+# Command to call when Docker is available
+DOCKER_MAIN_CMD="${CI_BUILD_DIR}/ci_build.sh"
+# Command to call when Docker is unavailable
+NO_DOCKER_MAIN_CMD="${CI_BUILD_DIR}/builds/configured"
+
+# Additional option flags to apply when Docker is unavailable (e.g., on Mac)
+NO_DOCKER_OPT_FLAG="-j 1 --genrule_strategy=standalone "\
```

Comment:
I've had trouble without --spawn_strategy=standalone but a lot has changed
since then.
On Thu, Feb 4, 2016 at 19:14 caisq <notifications@github.com> wrote:

> In tensorflow/tools/ci_build/ci_parameterized_build.sh
> <https://github.com/tensorflow/tensorflow/pull/998#discussion_r51972755>:
>
> > +str_strip () {
> > +  echo -e "$1" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//'
> > +}
> > +
> > +
> > +##########################################################
> > +# Default configuration
> > +CI_BUILD_DIR="tensorflow/tools/ci_build"
> > +
> > +# Command to call when Docker is available
> > +DOCKER_MAIN_CMD="${CI_BUILD_DIR}/ci_build.sh"
> > +# Command to call when Docker is unavailable
> > +NO_DOCKER_MAIN_CMD="${CI_BUILD_DIR}/builds/configured"
> > +
> > +# Additional option flags to apply when Docker is unavailable (e.g., on Mac)
> > +NO_DOCKER_OPT_FLAG="-j 1 --genrule_strategy=standalone "\
>
> I copied those from the mac build on Jenkins assuming they were necessary.
> I confirmed that except "--linkopt=-headerpad_max_install_names", the flags
> can be omitted without causing the build to fail. The "--linkopt" flag
> seems to be necessary. See:
> http://ci.tensorflow.org/job/tensorflow-master-mac/146/
>
> Keeping it ensures that both the clean and non-clean bazel build pass.
> http://ci.tensorflow.org/job/tensorflow-master-mac/147/
> http://ci.tensorflow.org/job/tensorflow-master-mac/148/
>
> So I'll remove the option flags except
> "--linkopt=-headerpad_max_install_names".
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/998/files#r51972755>.
>